foxtrot-server/src/main/java/com/flipkart/foxtrot/server/auth/authprovider/impl/IdmanAuthProvider.java

```diff
@@ -0,0 +1,177 @@
+                                                                IdType.ACCESS_TOKEN,
+                                                                TokenType.DYNAMIC,
+                                                                idmanUser.getUser().getId(),
+                                                                expiry));
+                               });
+    }
+
+    @Override
+    @SneakyThrows
+    public Optional<AuthenticatedInfo> authenticate(AuthInfo authInfo) {
+        return remoteTokenCall(
+                "refresh_token",
+                authInfo.accept(TokenAuthInfo::getToken),
+                ti -> {
+                    val idmanUser = ti.getUser();
+                    val expiry = Date.from(Instant.now().plusSeconds(ti.getExpiry()));
+                    val remoteUser = idmanUser.getUser();
+                    val t = new Token(ti.getAccessToken(),
+                                      IdType.ACCESS_TOKEN,
+                                      TokenType.DYNAMIC,
+                                      remoteUser.getId(),
+                                      expiry);
+                    val user = new User(remoteUser.getId(),
+                                        mapToFoxtrotRoles(idmanUser),
+                                        Collections.emptySet(),
+                                        isSystemUser(idmanUser),
+                                        new Date(),
+                                        new Date());
+                    return Optional.of(new AuthenticatedInfo(t, user));
+                });
+    }
+
+    @Override
+    @SneakyThrows
+    public boolean logout(String sessionId) {
+        if(Strings.isNullOrEmpty(sessionId)) {
+            log.warn("Empty token send for logout");
+            return true;
+        }
+        return remoteTokenRevokeCall(sessionId);
+    }
+
+    private Set<FoxtrotRole> mapToFoxtrotRoles(final IdmanUser idmanUser) {
+        switch (idmanUser.getRole()) {
+            case "FOXTROT_ADMIN": {
+                return EnumSet.allOf(FoxtrotRole.class);
+            }
+            case "FOXTROT_HUMAN_USER": {
+                return EnumSet.of(FoxtrotRole.CONSOLE, FoxtrotRole.QUERY, FoxtrotRole.INGEST);
+            }
+            case "FOXTROT_SYSTEM_USER": {
+                return EnumSet.of(FoxtrotRole.QUERY, FoxtrotRole.INGEST);
+            }
+            default:
+                throw new IllegalStateException("Unexpected value: " + idmanUser.getRole());
+        }
+    }
+
+    private boolean isSystemUser(final IdmanUser user) {
+        return user.getRole().equals("FOXTROT_ADMIN");
+    }
+
+    private <T> Optional<T> remoteTokenCall(
+            String callMode,
+            String param,
+            Function<TokenInfo, Optional<T>> handler) throws URISyntaxException, IOException {
+        val post = new HttpPost(new URIBuilder(config.getIdmanEndpoint())
+                                        .setPath("/apis/oauth2/token")
```

Comment:
It will be hosted separately for an org. Yes, it needs to be available for people to be able to login. Just like google needs to be up and working for people to login using google. Once authenticated, the auth info is cached for some time.